redash/query_runner/athena.py

```diff
@@ -130,28 +135,41 @@ def __get_schema_from_glue(self):
         database_paginator = client.get_paginator('get_databases')
         table_paginator = client.get_paginator('get_tables')
 
-        for databases in database_paginator.paginate():
-            for database in databases['DatabaseList']:
-                iterator = table_paginator.paginate(DatabaseName=database['Name'])
-                for table in iterator.search('TableList[]'):
-                    table_name = '%s.%s' % (database['Name'], table['Name'])
-                    if table_name not in schema:
-                        column = [columns['Name'] for columns in table['StorageDescriptor']['Columns']]
-                        schema[table_name] = {'name': table_name, 'columns': column}
-                        for partition in table.get('PartitionKeys', []):
-                            schema[table_name]['columns'].append(partition['Name'])
+        query = 'DatabaseList[*]'
+        if self.configuration.get('hide_other_schemas', False):
+            query = 'DatabaseList[?Name==`{}`]'.format(
+                self.configuration.get('schema', 'default'))
+
+        for database in database_paginator.paginate().search(query):
+            iterator = table_paginator.paginate(DatabaseName=database['Name'])
+            for table in iterator.search('TableList[]'):
+                table_name = '%s.%s' % (database['Name'], table['Name'])
+                if table_name not in schema:
+                    column = [columns['Name'] for columns in table['StorageDescriptor']['Columns']]
+                    schema[table_name] = {'name': table_name, 'columns': column}
+                    for partition in table.get('PartitionKeys', []):
+                        schema[table_name]['columns'].append(partition['Name'])
+
         return schema.values()
 
     def get_schema(self, get_stats=False):
         if self.configuration.get('glue', False):
             return self.__get_schema_from_glue()
 
         schema = {}
-        query = """
-        SELECT table_schema, table_name, column_name
-        FROM information_schema.columns
-        WHERE table_schema NOT IN ('information_schema')
-        """
+
+        if self.configuration.get('hide_other_schemas', False):
```

Comment:
See my previous comment, except that the conditional is well-balanced here :)

It would be nice if you could reduce the query text duplication here though.